scripts/templates/trcddi.cpp.mako

```diff
@@ -112,6 +116,10 @@ namespace ur_tracing_layer
             return result;
         }
 
+        // Recreate the logger in case env variables have been modified between
+        // program launch and the call to `urLoaderInit`
+        logger = std::move(logger::create_logger("tracing", true));
```

Comment:
I've done this because by default a user will need to set `UR_LOG_TRACING="level:info"` to enable tracing. To make the transition from SYCL_PI_TRACE smoother for users, this allows the SYCL runtime to provide an env var (e.g. `SYCL_UR_TRACE=1`) which acts as an alias for this. This involves programmatically setting the UR_LOG_TRACING env var, which means we have to recheck it after the global constructors for the layers, but before the layers are actually initialized.